### PR TITLE
Add release process docs to the Handbook

### PR DESCRIPTION
**Pull request by pento**
To maintain a central location for the release process documentation, this PR adds the docs to the handbook.

### CODE REVIEW

**Comment by pento** [comment]
@mtias: This includes a change to the existing process - the `Stable tag` header in `readme.txt` is updated to match the version. This ensures that the plugin zip file contains the version number, per [this request](https://wordpress.org/support/topic/missing-the-version-number-in-filename-of-zip-file/).

File: docs/release.md [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+## Writing the Release Post and Changelog
+
+* Open the [https://github.com/WordPress/gutenberg/pulls?page=2&q=is%3Apr+is%3Aclosed+sort%3Aupdated-desc](recently updated PRs view), and find the PR where the last version bump occurred.
+* Read through each PR since then, to determine if it needs to be included in the Release Post and/or changelog.
+* Choose a feature or two to highlight in the release post - record an animation of them in action.
+* Save the draft post on [https://make.wordpress.org/core/](make.wordpress.org/core), for publishing after the release.
+
+## Bumping the Version
+
+* Create a PR like [https://github.com/WordPress/gutenberg/pull/2370/files](this one), bumping the version number in `gutenberg.php` and `package.json`.
```
Comment (ntwb) [inline docs/release.md]: `package-lock.json` also needs to be bumped, see this more recent PR for reference: https://github.com/WordPress/gutenberg/pull/3479/files

Comment (pento) [reply docs/release.md]: Whee!

File: docs/release.md [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+* Edit `readme.txt` to include the changelog for the current release.
+* Add new files to the SVN repo, and remove old files, in the `trunk` directory:
+  Add new files: `svn st | grep '^\?' | awk '{print $2}' | svn add`
+  Delete old files: `svn st | grep '^!' | awk '{print $2}' | svn rm`
+* Commit the new version to `trunk`:
+  `svn ci -m "Committing version 1.x.0"`
+* Tag the new version. Change to the parent directory, and run:
+  `svn cp trunk tags/1.x.0`
+  `svn ci -m "Tagging version 1.x.0."`
+* Edit `trunk/readme.txt` to point to the new tag. The `Stable version` header in `readme.txt` should be updated to match the new release version number. After updating and committing that, the new version will be released:
```
Comment (mtias) [inline docs/release.md]: Just to be sure, we'll start doing this step as a separate commit?

**Review by mtias — APPROVED** [review]

File: docs/release.md (hunk 2 above)
Comment (pento) [reply docs/release.md]: Yup. You commit the new version to trunk, then tag the new version, then updated the `Stable version` tag.

**Pull request merged**